docs/testspec/src/12_pubkey_kem.rst

```diff
@@ -1,6 +1,124 @@
+   |                        |                                                                         |
+   |                        | * :srcref:`src/tests/data/pubkey/cmce_kat_hashed.vec`                   |
+   +------------------------+-------------------------------------------------------------------------+
+   | **Expected Output:**   | Above described test vector files contain expected values for:          |
+   |                        |                                                                         |
+   |                        | * Classic McEliece Public Key                                           |
+   |                        | * Classic McEliece Private Key                                          |
+   |                        | * Ciphertext                                                            |
+   |                        | * Shared Secret                                                         |
+   +------------------------+-------------------------------------------------------------------------+
+   | **Steps:**             | For each KAT vector:                                                    |
+   |                        |                                                                         |
+   |                        | #. Seed a AES-256-CTR-DRBG with the specified RNG seed                  |
+   |                        |                                                                         |
+   |                        | #. Use the seeded RNG to generate a Classic McEliece key pair and       |
+   |                        |    compare it to the expected public and private key in the test        |
+   |                        |    vector. This uses the key encoding as implemented in the reference   |
+   |                        |    implementation and hashes the result using SHAKE-256(512) to save    |
+   |                        |    disk space in the KAT vectors.                                       |
+   |                        |                                                                         |
+   |                        | #. Check that the expected algorithm properties of the generated key    |
+   |                        |    match the generic expectations for KEMs (supports key encapsulation, |
+   |                        |    reports a key strength in a reasonable interval, etc.).              |
+   |                        |                                                                         |
+   |                        | #. Extract the public key from the just generated key pair and compare  |
+   |                        |    it to the expected value in the test vector.                         |
+   |                        |                                                                         |
+   |                        | #. Encode both public and private key, and decode them again.           |
```

Comment:
How is that different to the prior step? Just to check en/decoding consistency?

Reply:
Yeah. Some algorithms apply DER encoding/decoding. Just checks that: 
`decoding(encoding(key)) == key`